functionaltests/utils_test.go

```diff
@@ -59,12 +60,36 @@ func copyTerraforms(t *testing.T) {
 	require.NoError(t, err)
 }
 
+type deploymentOutput struct {
+	// ElasticsearchURL holds the Elasticsearch URL.
+	ElasticsearchURL string
+
+	// Username holds the Elasticsearch username for basic auth.
+	Username string
+
+	// Password holds the Elasticsearch password for basic auth.
+	Password string
```

Comment:
This is true but not complete. `Username` and `Password` hold the superuser credentials generated by the cluster creation. As you correctly mentioned these are necessary for some API calls. I think it's worth to mention this.

```suggestion
	// Username holds the Elasticsearch super user username for basic auth.
	Username string

	// Password holds the Elasticsearch super user password for basic auth.
	Password string
```